vehicle-control/tools/sdr_compare.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2015, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+"""This script contains functions for collecting depth data
+and comparing SDRs encoded from the data. It is used to aid in
+assessing the quality of the depth encoder.
+
+Data is collected directly from unity by pressing "c" during game
+play while this script is running with collectDepthData().
+It may be necessary to press, hold, and then release the key until
+"Collecting data." is printed. This will save the depth data as a numpy
+array in a file titled with the datetime. A screenshot will also be saved
+with the datetime in vehicle-control/simulations/Screenshots/. Compare
+the datetime of both files to ensure that screenshots and depth data were
+collected.
+
+To pairwise compare all the n encodings of saved depth data, run
+compareDepthData(). This will generate a nxn plot where the bottom left
+triangle depicts the SDRs overlapped and the top right triangle displays
+the percent overlap (relative to the density).
+"""
+
+
+import time
+import matplotlib.cm as cm
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+
+from unity_client.fetcher import Fetcher
+from sensorimotor.encoders.one_d_depth import OneDDepthEncoder
+
+positions = [i*10 for i in range(18, 54)]
+nPerPosition = 28
+
+_SHAPE = (36, nPerPosition)
+_OUTPUT_DIR = "depths/"
```

Comment:
This doesn't work unless there's already a `depths` directory. Either create one and commit it, or have this script create one if it doesn't exist.